app/src/main/java/de/danoeh/antennapod/fragment/preferences/NetworkPreferencesFragment.java

```diff
@@ -174,4 +152,145 @@ private void showUpdateIntervalTimePreferencesDialog() {
+
+        Spinner spinner = (Spinner) v_iew.findViewById(R.id.spinner);
+        RadioButton radioButton_interval = (RadioButton) v_iew.findViewById(R.id.intervalRadioButton);
+        RadioButton radioButtonTime = (RadioButton) v_iew.findViewById(R.id.timeRadioButton);
+        RadioButton radioButton_disable = (RadioButton) v_iew.findViewById(R.id.disableButton);
+        RadioGroup radioGroupOne = (RadioGroup) v_iew.findViewById(R.id.genderRadioGroup);
+        TimePicker timePicker = (TimePicker) v_iew.findViewById(R.id.timePicker);
+
+
+
+        // ->Spinner
+        //Get default Interval Values from Array
+        final String[] values = context.getResources().getStringArray(R.array.update_intervall_values);
+        //Get from Preferences Array in String
+        final String[] entries = getUpdateIntervalEntries(values);
+        //Get Interval Value from User Preferences
+        long currInterval = UserPreferences.getUpdateInterval();
+        int checkedItem = -1;
+        if(currInterval > 0)
+        {
+            String currIntervalStr = String.valueOf(TimeUnit.MILLISECONDS.toHours(currInterval));
+            checkedItem = ArrayUtils.indexOf(values, currIntervalStr);
+        }
+        ArrayAdapter<String> spinnerArrayAdapter = new ArrayAdapter<String>(getContext(), android.R.layout.simple_spinner_item, entries); //selected item will look like a spinner set from XML
+        spinnerArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+
+        //TIMER
+        int hourOfDay = 7;
+        int minute = 0;
+        int[] updateTime = UserPreferences.getUpdateTimeOfDay();
+        if (updateTime.length == 2) {
+            hourOfDay = updateTime[0];
+            minute = updateTime[1];
+        }
+
+
+
+
+        int finalCheckedItem = checkedItem;
+        int finalHourOfDay = hourOfDay;
+        int finalMinute = minute;
+        radioGroupOne.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(RadioGroup radioGroup, int i)
+            {
+                switch (i) {
+                    case R.id.intervalRadioButton:
+
+                        spinner.setAdapter(spinnerArrayAdapter);
+                        spinner.setSelection(finalCheckedItem);
+                        spinner.setVisibility(View.VISIBLE);
+                        timePicker.setVisibility(View.GONE);
+                        string_spinnercase[0] = R.string.pref_autoUpdateIntervallOrTime_CheckStatusInterval;
+                        break;
+                    case R.id.timeRadioButton:
+                        spinner.setVisibility(View.GONE);
+                        timePicker.setVisibility(View.VISIBLE);
+                        string_spinnercase[0] = R.string.pref_autoUpdateIntervallOrTime_CheckStatusTime;
+
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M)
+                        {
```

Comment:
Please have a look at the google code style. We use brackets in the same line. Like this:

```suggestion
                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```